vendor/google_riscv-dv/scripts/instr_trace_compare.py

```diff
@@ -124,45 +137,47 @@ def compare_trace_csv(csv1, csv2, name1, name2,
-                        gpr_trace_1[gpr][trace_1_index].get_trace_string()))
-                  print("%s[%d] : %s" % (name2, trace_2_index,
-                        gpr_trace_2[gpr][trace_2_index].get_trace_string()))
+                  fd.write("Mismatch:\n")
+                  fd.write("%s[%d] : %s\n" % (name1, trace_1_index,
+                           gpr_trace_1[gpr][trace_1_index].get_trace_string()))
+                  fd.write("%s[%d] : %s\n" % (name2, trace_2_index,
+                           gpr_trace_2[gpr][trace_2_index].get_trace_string()))
                 trace_2_index += 1
               else:
                 if verbose:
-                  print("Skipping %s[%d] : %s" %
-                        (name1, trace_1_index,
-                         gpr_trace_1[gpr][trace_1_index].get_trace_string()))
+                  fd.write("Skipping %s[%d] : %s\n" %
+                           (name1, trace_1_index,
+                           gpr_trace_1[gpr][trace_1_index].get_trace_string()))
                 coalesced_updates += 1
             else:
               coalesced_updates = 0
               matched_cnt += 1
               if verbose:
-                print("Matched [%0d]: %s : %s" %
-                      (trace_1_index, name1,
-                       gpr_trace_1[gpr][trace_1_index].get_trace_string()))
+                fd.write("Matched [%0d]: %s : %s\n" %
+                         (trace_1_index, name1,
+                         gpr_trace_1[gpr][trace_1_index].get_trace_string()))
               trace_2_index += 1
       # Check the final value match between the two traces
       for gpr in gpr_trace_1:
         if (len(gpr_trace_1[gpr]) == 0 or len(gpr_trace_2[gpr]) == 0):
           mismatch_cnt += 1
-          print("Zero GPR[%s] updates observed: %s:%d, %s:%d" % (gpr,
-                name1, len(gpr_trace_1[gpr]), name2, len(gpr_trace_2[gpr])))
+          fd.write("Zero GPR[%s] updates observed: %s:%d, %s:%d\n" % (gpr,
+                   name1, len(gpr_trace_1[gpr]), name2, len(gpr_trace_2[gpr])))
         elif long(gpr_trace_1[gpr][-1].rd_val, 16) != \
              long(gpr_trace_2[gpr][-1].rd_val, 16):
           mismatch_cnt += 1
           if mismatch_cnt <= mismatch_print_limit:
-            print("Mismatch final value:")
-            print("%s : %s" % (name1, gpr_trace_1[gpr][-1].get_trace_string()))
-            print("%s : %s" % (name2, gpr_trace_2[gpr][-1].get_trace_string()))
+            fd.write("Mismatch final value:\n")
+            fd.write("%s : %s\n" % (name1, gpr_trace_1[gpr][-1].get_trace_string()))
+            fd.write("%s : %s\n" % (name2, gpr_trace_2[gpr][-1].get_trace_string()))
     if mismatch_cnt == 0:
       compare_result = "PASSED"
     else:
       compare_result = "FAILED"
-    print("Compare result[%s]: %d matched, %d mismatch" %
-          (compare_result, matched_cnt, mismatch_cnt))
+    fd.write("Compare result[%s]: %d matched, %d mismatch\n\n" %
```

Comment:
is the post_compare step no longer going to display any terminal output? I feel that maybe printing the overall regression result might be helpful

Reply:
- fd is an optional input, if it's omitted, the output will be redirected to stdout
- Right now there will be no terminal output for each compare result(it can be added), but the regression summary will be displayed on the terminal

Reply:
Sounds good